### DISCUSSION TITLE
Provide an Admin Panel

### DISCUSSION THREAD

**Discussion by ducdetronquito** [loco-rs/loco#167] [Ideas]
Hello !

First, thank you for the work you are doing on Loco; I haven't used it yet but I plan to give it a try during the christmas holidays :)

I'm wondering if you envision a built-in panel admin for Loco, like the one we can find in [Django ](https://docs.djangoproject.com/en/5.0/ref/contrib/admin/) (or in [Rails](https://activeadmin.info/), or in [Laravel](https://orchid.software/), etc...)

I work with Django every day and having an easy way to hack a CRUD app is a blessing, either for testing, administrating a backend service or even expose it as a UI for clients (internal or external).

Have a great day !

**Comment by kaplanelad** [comment]
Hey @ducdetronquito, great suggestion!  But right now, it's not our primary focus.

You're more than welcome to contribute this feature. If you're interested, we can talk about the design more. Let me know if you'd like to dive into the details!

**Reply by ducdetronquito** [reply]
Thanks for the quick answer @kaplanelad  !
I totally understand that this is not your primary focus.

I keep that in mind, first I want to toy with Loco ^^

**Comment by aJuvan** [comment]
I'd love see this feature in Loco as well and I'm planning to contribute something like this to the project.

I'm partial to the Django approach where the admin module is optional and once enabled is really easy to set up with just a few decorators and some class metadata. But I've been also looking into the React-admin project where you can just create a provider compatible API (or API and a provider) to configure it.

With either of these approaches the goal is having a minimal user configuration for the majority of use cases. If anyone has any comments about which of these two approaches would work better for Loco (or if you suggest another approach) do tell so we can get the most out of it. This goes for special features you'd like to have as well.

**Comment by mustafasegf** [comment]
I saw this interesting project https://github.com/mgugger/actix-admin, if you permit I can port that into axum and integrate that to loco

**Reply by problem** [reply]
That would be amazing!
